src/app/app.component.ts

```diff
@@ -91,11 +91,17 @@ export class AppComponent implements OnDestroy, OnInit {
       this.ngZone.run(async () => {
         switch (message.command) {
           case "loggedIn":
+            this.notificationsService.updateConnection(false);
+            break;
           case "loggedOut":
+            this.routerService.setPreviousUrl(null);
+            this.notificationsService.updateConnection(false);
+            break;
           case "unlocked":
             this.notificationsService.updateConnection(false);
             break;
           case "authBlocked":
+            window.setTimeout(() => this.routerService.setPreviousUrl(message.url), 500);
```

Comment:
Did this just feel like a reasonable timeout, or are we trying to slow things down intentionally with this? I believe we just need this to finish prior to `/` submitting, right?

Reply:
We have the same timeout a couple of lines below. I couldn't be bothered to figure out why. 😿